pkg/rest/endpoints/encoders.go

```diff
@@ -37,41 +42,82 @@ func encodeApplicationErrors(_ context.Context, w http.ResponseWriter, response
-	if _, ok := response.(middleware.AppError); ok {
-		return encodeApplicationErrors(context, w, response)
+func NewEncodeJsonResponse(returnCode int) kithttp.EncodeResponseFunc {
+	return func(context context.Context, w http.ResponseWriter, response interface{}) error {
+		if _, ok := response.(middleware.AppError); ok {
+			return encodeApplicationErrors(context, w, response)
+		}
+		return encodeJsonResponse(w, response, returnCode)
 	}
-	return encodeJsonResponse(w, response, http.StatusCreated)
 }
 
-func EncodeGetResponse(context context.Context, w http.ResponseWriter, response interface{}) error {
-	if _, ok := response.(middleware.AppError); ok {
-		return encodeApplicationErrors(context, w, response)
+func NewEncodeINIResponse(returnCode int) kithttp.EncodeResponseFunc {
+	return func(context context.Context, w http.ResponseWriter, response interface{}) error {
+		if _, ok := response.(middleware.AppError); ok != false {
+			return encodeApplicationErrors(context, w, response)
+		}
+		return encodeINIResponse(w, response, returnCode)
 	}
-	return encodeJsonResponse(w, response, http.StatusOK)
 }
 
-func EncodePlainTextGetResponse(context context.Context, w http.ResponseWriter, response interface{}) error {
-	if _, ok := response.(middleware.AppError); ok != false {
-		return encodeApplicationErrors(context, w, response)
+func NewEncodeYamlResponse(returnCode int) kithttp.EncodeResponseFunc {
+	return func(context context.Context, w http.ResponseWriter, response interface{}) error {
+		if _, ok := response.(middleware.AppError); ok != false {
+			return encodeApplicationErrors(context, w, response)
+		}
+		return encodeYamlResponse(w, response, returnCode)
 	}
-	w.Header().Set("Content-Type", "text/plain")
-	w.WriteHeader(http.StatusOK)
-	_, err := w.Write([]byte(response.(string)))
-	return err
 }
 
-func EncodeDeleteResponse(context context.Context, w http.ResponseWriter, response interface{}) error {
-	return EncodeGetResponse(context, w, response)
+func encodeJsonResponse(w http.ResponseWriter, response interface{}, returnCode int) error {
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(returnCode)
+	return json.NewEncoder(w).Encode(response)
 }
 
-func EncodePutResponse(context context.Context, w http.ResponseWriter, response interface{}) error {
-	return EncodeGetResponse(context, w, response)
+func encodeYamlResponse(w http.ResponseWriter, response interface{}, returnCode int) error {
+	w.Header().Set("Content-Type", "application/yaml")
+	w.WriteHeader(returnCode)
+	b, err := yaml.Marshal(response)
+	if err != nil {
+		return err
+	}
+	_, err = w.Write(b)
+	return err
 }
 
-func encodeJsonResponse(w http.ResponseWriter, response interface{}, returnCode int) error {
-	w.Header().Set("Content-Type", "application/json")
+func encodeINIResponse(w http.ResponseWriter, response interface{}, returnCode int) error {
+	w.Header().Set("Content-Type", "text/plain")
```

Comment:
I don't think so, but we should add some constants for these types. Will fix that. I think go-restful has only two, one for JSON and one for XML. Will add one for Yaml and plain text.